lang/cpp.go

```diff
@@ -0,0 +1,470 @@
+				functionCall := fmt.Sprintf(
+					"%s->%s(%s)",
+					objectName,
+					method.Name,
+					c.getParamString(method.Params),
+				)
+				if returnType != "void" {
+					callCode += fmt.Sprintf(
+						"\t\t\t%s %s << ',';\n\t\t}",
+						c.getPrintCodeForType(dimCnt, returnType, functionCall, outputFileStreamName),
+						outputFileStreamName,
+					)
+				} else {
+					callCode += fmt.Sprintf(
+						"\t\t\t%s;\n\t\t\t%s << \"null,\";\n\t\t}",
+						functionCall,
+						outputFileStreamName,
+					)
+				}
+			}
+			callCode += fmt.Sprintf(
+				" else {\n\t\t\treturn 1;\n\t\t}\n\t\t%s.ignore();\n",
+				inputFileStreamName,
+			)
+		}
+		callCode += fmt.Sprintf(
+			"\t}\n\t%s.seekp(-1, ios_base::end); %s << ']';\n",
+			outputFileStreamName,
+			outputFileStreamName,
+		)
+	} else {
+		callCode = fmt.Sprintf(
+			"\tauto &&%s = %s->%s(%s);\n",
+			returnName,
+			objectName,
+			q.MetaData.Name,
+			c.getParamString(q.MetaData.Params),
+		)
+	}
+	return callCode
+}
+
+func (c cpp) generatePrintCode(q *leetcode.QuestionData) string {
+	if q.MetaData.SystemDesign {
+		return ""
+	}
+	dimCnt, cppType := c.getCppTypeName(q.MetaData.Return.Type)
+	return fmt.Sprintf("\t%s\n", c.getPrintCodeForType(dimCnt, cppType, returnName, outputFileStreamName))
+}
+
+func (c cpp) generateTest(q *leetcode.QuestionData, testcases string) string {
+	content := fmt.Sprintf(testFileHeader, c.lineComment)
+	content += fmt.Sprintf(
+		cppTestFileTemplate,
+		structDefi,
+		helperFuncs,
+		c.generateScanCode(q),
+		c.generateInitCode(q),
+		c.generateCallCode(q),
+		c.generatePrintCode(q),
+	)
+	return content
+}
+
+func (c cpp) getJudgeResult(dimCnt int, returnType string, expectedOutput string, actualOutput string) (eq bool) {
+	// type specific judge
+	//  * double - not needed at this time, as all double results are sanitized as "%.5f"
+	//  * integer, long, string - raw string comparison is enough
+	/*if returnType == "double" {
+		eps := 1e-5
+		if dimCnt > 0 {
+			return false
+		}
+		a, _ := strconv.ParseFloat(expectedOutput, 32)
+		b, _ := strconv.ParseFloat(actualOutput, 32)
+		eq = math.Abs(a-b) < eps
+	} else*/{
+		eq = expectedOutput == actualOutput
+	}
+	return
+}
+
+func (c cpp) RunLocalTest(q *leetcode.QuestionData, dir string) (bool, error) {
+	red := color.New(color.FgRed).Add(color.Bold)
+	green := color.New(color.FgGreen).Add(color.Bold)
+	magenta := color.New(color.FgMagenta).Add(color.Bold)
+	yellow := color.New(color.FgYellow).Add(color.Bold)
+	blue := color.New(color.FgBlue).Add(color.Bold)
+
+	// get path
+	filenameTmpl := getFilenameTemplate(q, c)
+	baseFilename, err := q.GetFormattedFilename(c.slug, filenameTmpl)
+	if err != nil {
+		return false, err
+	}
+	testFile := filepath.Join("cpp/", baseFilename, "solution.cpp")
+	execFile := filepath.Join("cpp/", baseFilename, "solution.exec")
+	inputFile := filepath.Join("cpp/", baseFilename, "input.txt")
+	outputFile := filepath.Join("cpp/", baseFilename, "output.txt")
```

Comment:
should not use `cpp/` here, should use `dir` variable instead

Reply:
will fix later today